### PR TITLE
Support to retrieve data from ListConfigFiles webmethod using SAP Management Console

### PR DESCRIPTION
**Pull request by defensahacker**
Minor changes to existent SAP scanner modules to retrieve data from the ListConfigFiles webmethod.

This module attempts to list the config files through the SAP Management Console SOAP Interface.
Returns a list of config files found in the SAP configuration with its absolute paths inside the server filesystem.

### CODE REVIEW

File: modules/auxiliary/scanner/sap/sap_mgmt_con_listconfigfiles.rb [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+    if not res
+      print_error("#{rhost}:#{rport} [SAP] Unable to connect")
+      return
+    end
+
+    enum_instance(ip)
+  end
+
+  def enum_instance(rhost)
```
Comment (bcoles) [inline modules/auxiliary/scanner/sap/sap_mgmt_con_listconfigfiles.rb]: I realise this is copypasta from the other SAP module, but the `enum_instance` method could be cleaned up as follows:

```ruby
  def enum_instance(rhost)
    print_status("#{rhost}:#{rport} [SAP] Connecting to SAP Management Console SOAP Interface")

    soapenv = 'http://schemas.xmlsoap.org/soap/envelope/'
    xsi = 'http://www.w3.org/2001/XMLSchema-instance'
    xs = 'http://www.w3.org/2001/XMLSchema'
    sapsess = 'http://www.sap.com/webas/630/soap/features/session/'
    ns1 = 'ns1:ListConfigFiles'

    data = '<?xml version="1.0" encoding="utf-8"?>' + "\r\n"
    data << '<SOAP-ENV:Envelope xmlns:SOAP-ENV="' + soapenv + '"  xmlns:xsi="' + xsi
    data << '" xmlns:xs="' + xs + '">' + "\r\n"
    data << '<SOAP-ENV:Header>' + "\r\n"
    data << '<sapsess:Session xlmns:sapsess="' + sapsess + '">' + "\r\n"
    data << '<enableSession>true</enableSession>' + "\r\n"
    data << '</sapsess:Session>' + "\r\n"
    data << '</SOAP-ENV:Header>' + "\r\n"
    data << '<SOAP-ENV:Body>' + "\r\n"
    data << '<' + ns1 + ' xmlns:ns1="urn:SAPControl"></' + ns1 + '>' + "\r\n"
    data << '</SOAP-ENV:Body>' + "\r\n"
    data << '</SOAP-ENV:Envelope>' + "\r\n\r\n"

    res = send_request_raw({
      'uri'      => normalize_uri(datastore['URI']),
      'method'   => 'POST',
      'data'     => data,
      'headers'  =>
        {
          'Content-Length' => data.length,
          'SOAPAction'     => '""',
          'Content-Type'   => 'text/xml; charset=UTF-8',
        }
    }, 15)

    unless res
      print_error("#{rhost}:#{rport} [SAP] Unable to connect")
      return
    end

    if res.code == 500 && res.body =~ %r{<faultstring>(.*)</faultstring>}i
      print_error("#{rhost}:#{rport} [SAP] Error code: #{$1.strip}")
      return
    end

    if res.code == 200 && res.body =~ %r{<item>([^<]+)</item>}i
      env = res.body.scan(%r{<item>([^<]+)</item>}i)
      print_good("#{rhost}:#{rport} [SAP] List of Config Files")
      env.each do |output|
        print_good(output.first)
      end
      return
    end

    print_error("#{rhost}:#{rport} [SAP] Failed to identify instance properties")
  rescue ::Rex::ConnectionError
    print_error("#{rhost}:#{rport} [SAP] Unable to connect")
  end
```

**Comment by bcoles** [comment]
It is required that code in your fork be merged **from a unique branch** in your repository to master in Rapid7's.  Please create a new branch in your fork of framework and resubmit this from that branch.

```
git checkout -b <BRANCH_NAME>
git push <your_fork_remote> <BRANCH_NAME>
```

This helps protect the process, ensure users are aware of commits on the branch being considered for merge, allows for a location for more commits to be offered without mingling with other contributor changes and allows contributors to make progress while a PR is still being reviewed.

Closing based on the this requirement, please do resubmit from a unique branch.

**Comment by spinfoo** [comment]
Awesome! Thanks for the feedback

**Comment by bcoles** [comment]
#11282

**Pull request closed without merging**